qcfractal/interface/collections/reaction_dataset.py

```diff
@@ -239,16 +372,120 @@ def visualize(self,
 
         return self._visualize(metric, bench, query=query, groupby=groupby, return_figure=return_figure, kind=kind)
 
+    def get_molecules(self, subset: Optional[Union[str, Set[str]]] = None,
+                      stoich: Union[str, List[str]] = "default") -> Union[pd.DataFrame, 'Molecule']:
+        """Queries full Molecules from the database.
+
+        Parameters
+        ----------
+        subset : Optional[Union[str, Set[str]]], optional
+            The index subset to query on
+        stoich : Union[str, List[str]], optional
+            The stoichiometries to pull from, either a single or multiple stoichiometries
+
+        Returns
+        -------
+        Union[pd.DataFrame, 'Molecule']
+            Either a DataFrame of indexed Molecules or a single Molecule if a single subset string was provided.
+        """
+
+        self._check_client()
+        self._check_state()
+        self._validate_stoich(stoich)
+
+        indexer, names = self._molecule_indexer(stoich, subset)
+        df = self._get_molecules(indexer)
+        df.index = pd.MultiIndex.from_tuples(df.index, names=names)
+
+        if isinstance(subset, str) and isinstance(stoich, str):
+            return df.iloc[0, 0]
+        else:
+            return df
+
+    def get_records(self,
+                    method,
+                    basis: Optional[str] = None,
+                    *,
+                    keywords: Optional[str] = None,
+                    program: Optional[str] = None,
+                    stoich: Union[str, List[str]] = "default",
+                    projection: Optional[Dict[str, bool]] = None,
+                    subset: Optional[Union[str, Set[str]]] = None) -> Union[pd.DataFrame, 'ResultRecord']:
+        """
+        Queries the local Portal for the requested keys and stoichiometry.
+
+        Parameters
+        ----------
+        method : str
+            The computational method to query on (B3LYP)
+        basis : Optional[str], optional
+            The computational basis to query on (6-31G)
+        keywords : Optional[str], optional
+            The option token desired
+        program : Optional[str], optional
+            The program to query on
+        stoich : Union[str, List[str]], optional
+            The given stoichiometry to compute.
+        projection : Optional[Dict[str, bool]], optional
+            The attribute project to perform on the query, otherwise returns ResultRecord objects.
+        subset : Optional[Union[str, Set[str]]], optional
```

Comment:
The subset keyword does not appear to work:

```
s22.get_records("b3lyp", subset="Water Dimer").reset_index()["name"]
```
gives:
```
0     2-Pyridone-2-Aminopyridine Complex
1     2-Pyridone-2-Aminopyridine Complex
2     2-Pyridone-2-Aminopyridine Complex
3     2-Pyridone-2-Aminopyridine Complex
4          Adenine-Thymine Complex Stack
                     ...                
83                    Uracil Dimer Stack
84                           Water Dimer
85                           Water Dimer
86                           Water Dimer
87                           Water Dimer
Name: name, Length: 88, dtype: object
```

I think this is because `subset` is only used as an argument to `_molecule_indexer`, and `_molecule_indexer` ignores `subset`.